_docs/pages/instrumentation/common-use-cases.md

```diff
@@ -253,3 +253,62 @@ if request.get('debug'):
         tags={tags.SAMPLING_PRIORITY: 1}
     )
 ```
+
+### Tracing Message Bus Scenarios
+
+There are two message bus styles that should be handled, Message Queues and Publish/Subscribe (Topics).
+
+From a tracing perspective, the message bus style is not important, only that the span context associated with the producer is propagated to the zero or more consumers of the message. It is then the responsibility of the consumer(s) to create a span to encapsulate processing of the consumed message and establish a _FollowsFrom_ reference to the propagated span context.
+
+As with the RPC client example, a messaging producer is expected to start a new tracing Span before sending a message, and propagate the new Span along with that message. The following example shows how it can be done.
+
+```python
+def traced_send(message, operation):
+    # retrieve current span from propagated message context
+    parent_span = get_current_span()
+
+    # start a new span to represent the message producer
+    span = tracer.start_span(
+        operation_name=operation,
+        child_of=parent_span.context,
+        tags={'message.destination': message.destination}
+    )
+
+    # propagate the Span via message headers
+    tracer.inject(
+        span.context,
+        format=opentracing.TEXT_MAP_FORMAT,
+        carrier=message.headers)
+
+    with span:
+        messaging_client.send(message)
+    except Exception e:
+        span.log(event='general exception', payload=e)
```

Comment:
there is a scheme for error logging... https://github.com/opentracing/specification/blob/master/semantic_conventions.md#log-fields-table

I'm also fine leaving this out of the example entirely since it's a bit of a distraction.